autonomi/src/client/data_types/scratchpad.rs

```diff
@@ -292,12 +292,16 @@ impl Client {
         let current = match self.scratchpad_get(&address).await {
             Ok(scratchpad) => Some(scratchpad),
             Err(ScratchpadError::GetError(GetError::RecordNotFound)) => None,
+            // forks should not stop updates as updates are here to resolve forks, hence the max_by_key
             Err(ScratchpadError::GetError(GetError::Network(NetworkError::SplitRecord(
                 result_map,
             )))) => result_map
                 .values()
                 .filter_map(|record| try_deserialize_record::<Scratchpad>(record).ok())
                 .max_by_key(|scratchpad: &Scratchpad| scratchpad.counter()),
+            Err(ScratchpadError::Fork(scratchpads)) => scratchpads
```

Comment:
the `scratch_get` (same as to `pointer_get`) already carried out `resolve_split_records` , which means the Fork returned here being the only situation of `same counter with different content`, hence `max_by_key` won't resolve (or resolve by mistake, just picking the first entry of the highest counter)